Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/TaskContainerFragment.java

```diff
@@ -126,22 +150,81 @@ private void handleAssetSelecting(Intent data, AssetType assetType) {
             Long assetId = assetRepository
                     .create(AssetType.getTypeByExtension(assetName), assetName);
             setAssetValue(assetType, assetName, assetId);
-
         } catch (IOException e) {
-            String pickingFileError = getResources().getString(R.string.picking_file_error);
-            Toast errorToast = Toast.makeText(context, pickingFileError, Toast.LENGTH_LONG);
-            errorToast.show();
+            showToastMessage(R.string.picking_file_error);
         }
     }
 
     private void setAssetValue(AssetType assetType, String assetName, Long assetId) {
+
+        StringBuilder sb = new StringBuilder(assetName);
+        String croppedName = sb.replace(assetName.lastIndexOf("_"), assetName.lastIndexOf("."), "")
+                .toString();
+
         if (assetType.equals(AssetType.PICTURE)) {
-            taskPicture.setText(assetName);
+            taskPicture.setText(croppedName);
             pictureId = assetId;
         } else {
-            taskSound.setText(assetName);
+            taskSound.setText(croppedName);
             soundId = assetId;
         }
     }
+
+    private void playStopSound() {
+        boolean isNameEmpty = taskSound.getText().toString().isEmpty();
+        if (!isNameEmpty) {
+            if (!mp.isPlaying()) {
+                playSound(retrieveSoundFile());
+                runBtnAnimation();
+            } else {
+                stopSound();
+                stopBtnAnimation();
+            }
+        } else {
+            showToastMessage(R.string.no_file_to_play_error);
+        }
+    }
+
+    private void playSound(String pathToFile) {
+        try {
+            mp.reset();
+            mp.setDataSource(pathToFile);
+            mp.prepare();
+            mp.start();
+        } catch (IOException e) {
+            showToastMessage(R.string.playing_file_error);
+        }
+    }
+
+    private String retrieveSoundFile() {
```

Comment:
Mikolay, we were talking that I have to extract some method from your AssestHelper class, to use it to build a path to file. But I didn't find anything worth. Maybe I missed smth. 

Also, I don't required to @inject variables. Everything I took from existing fields. Please review.